src/Send/AssetSelectorScreen/AssetSelectorScreen.tsx

```diff
@@ -10,28 +11,40 @@ import NoImage from '../../assets/img/asset_no_image.png'
 import {Boundary, Button, Spacer, Text, TextInput} from '../../components'
 import {useTokenInfo} from '../../hooks'
 import globalMessages, {txLabels} from '../../i18n/global-messages'
+import {getDefaultAssetByNetworkId} from '../../legacy/config'
 import {decodeHexAscii, formatTokenAmount, getAssetDenominationOrId, getTokenFingerprint} from '../../legacy/format'
 import {useSelectedWallet} from '../../SelectedWallet'
 import {COLORS} from '../../theme'
-import {Token, TokenEntry} from '../../types'
+import {Token} from '../../types'
+import {YoroiWallet} from '../../yoroi-wallets'
+import {Quantity, TokenId, YoroiAmounts} from '../../yoroi-wallets/types'
+import {Quantities} from '../../yoroi-wallets/utils'
 
 type Props = {
-  assetTokens: Array<TokenEntry>
-  onSelect: (tokenEntry: TokenEntry) => void
+  balances: YoroiAmounts
+  onSelect: (tokenId: TokenId) => void
   onSelectAll: () => void
 }
 
-export const AssetSelectorScreen = ({assetTokens, onSelect, onSelectAll}: Props) => {
+export const AssetSelectorScreen = ({balances, onSelect, onSelectAll}: Props) => {
   const strings = useStrings()
-
+  const wallet = useSelectedWallet()
+  const defaultAsset = getDefaultAssetByNetworkId(wallet.networkId)
   const [matcher, setMatcher] = React.useState('')
   const onChangeMatcher = (matcher: string) => {
     LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
     setMatcher(matcher)
   }
-  const sortedAssetTokens = assetTokens
-    .sort((a: TokenEntry, b: TokenEntry) => (a.amount.isGreaterThan(b.amount) ? -1 : 1))
-    .sort((assetToken) => (assetToken.identifier === '' ? -1 : 1))
+
+  const sortedBalance: Array<[TokenId, Quantity]> = useMemo(
+    () =>
+      Object.entries(balances)
+        .sort(([, quantityA]: [TokenId, Quantity], [, quantityB]: [TokenId, Quantity]) =>
+          Quantities.isGreaterThan(quantityA, quantityB) ? -1 : 1,
```

Comment:
Quantities.diff

Reply:
After that we need to transform to `number` right?

Reply:
Done